lib/setup/setupInstall.js

```diff
@@ -754,34 +754,38 @@ function Install(options) {
             }
         }
 
-        // install linux/osx libraries
-        packetManager = packetManager || new PacketManager();
-        packetManager.install(adapterConf.common.osDependencies && adapterConf.common.osDependencies[process.platform])
-            .catch(e => console.error(e.message))
-            .then(() => {
-                if (!fs.existsSync(adapterDir + '/node_modules')) {
-                    // Install node modules
-                    installNpm(adapter, (err, _adapter) => {
-                        if (err) {
-                            processExit(err);
-                        } else {
-                            upload.uploadAdapter(_adapter, true, true, null, null, () =>
-                                upload.uploadAdapter(_adapter, false, true, null, null,() =>
-                                    callInstallOfAdapter(_adapter, adapterConf, () =>
-                                        that.uploadStaticObjects(adapter, _err =>
-                                            upload.upgradeAdapterObjects(adapter, () =>
-                                                callback(adapter))))));
-                        }
-                    });
+        if (adapterConf.common.osDependencies && adapterConf.common.osDependencies[process.platform]) {
+            // install linux/osx libraries
+            try {
+                packetManager = packetManager || new PacketManager();
+                await packetManager.install(adapterConf.common.osDependencies[process.platform]);
+            } catch (e) {
+                adapter.log.error(`Could not install required OS packages: ${e.message}`);
+            }
+        }
+
+        if (!fs.existsSync(adapterDir + '/node_modules')) {
+            // Install node modules
+            installNpm(adapter, (err, _adapter) => {
+                if (err) {
+                    processExit(err);
                 } else {
-                    upload.uploadAdapter(adapter, true, true, () =>
```

Comment:
unsure here about the rename ...

Reply:
this is now line https://github.com/ioBroker/ioBroker.js-controller/pull/1223/files#diff-6e78e9d26877a71fa3c3f7fa44eb257d46f727725a95f42f16d87ca036499afaR782 just diff a little bit screwed